### PR TITLE
Automated cherry pick of #3075: Avoid endless reconcile loop when image pull secrets are

### PR DESCRIPTION
**Pull request by rene-dekker**
Cherry pick of #3075 on release-v1.33.

#3075: Avoid endless reconcile loop when image pull secrets are

### CODE REVIEW

**Review by caseydavenport — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
